frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -918,9 +911,92 @@ pub mod pallet {
 
             Ok(().into())
         }
+
+        /// Used to burn unclaimed & stale rewards from unregistered contract
+        #[pallet::weight(T::WeightInfo::claim_dapp())]
+        pub fn burn_stale_reward(
+            origin: OriginFor<T>,
+            contract_id: T::SmartContract,
+            #[pallet::compact] era: EraIndex,
+        ) -> DispatchResultWithPostInfo {
+            Self::ensure_pallet_enabled()?;
+            ensure_root(origin)?;
+
+            let dapp_info =
+                RegisteredDapps::<T>::get(&contract_id).ok_or(Error::<T>::NotOperatedContract)?;
+            ensure!(
+                dapp_info.is_unregistered(),
+                Error::<T>::NotUnregisteredContract
+            );
+
+            let current_era = Self::current_era();
+
+            let burn_era_limit =
+                current_era.saturating_sub(T::UnregisteredDappRewardRetention::get());
+            ensure!(era < burn_era_limit, Error::<T>::EraOutOfBounds);
+
+            let mut contract_stake_info =
+                Self::contract_stake_info(&contract_id, era).unwrap_or_default();
+
+            let dapp_reward = Self::calculate_dapp_reward(&contract_stake_info, &dapp_info, era)?;
+
+            // Withdraw reward funds from the dapps staking pot and burn them
+            let imbalance_to_burn = T::Currency::withdraw(
+                &Self::account_id(),
+                dapp_reward,
+                WithdrawReasons::TRANSFER,
+                ExistenceRequirement::AllowDeath,
+            )?;
+            mem::drop(imbalance_to_burn);
```

Comment:
No, it's not needed explicitly.
I tested before without it and it worked just fine.

But I wanted to be explicit so it's more clear what I'm doing.
I also found the `burn_from` functionality but had some issues with it. Will look into it again.

Reply:
`let _ = T::Currency::withdraw(`

Should works the same.

Reply:
Yes, I implemented it initially almost like that.
It was `let _reward_to_burn = ...` :slightly_smiling_face: 

I added the `drop` later since I wanted to be more explicit.
We can change it in future PR, I don't really mind.